### PR TITLE
Adds react context provider and nextjs example

### PR DESCRIPTION
**Pull request by taylorjdawson**
### Checklist
- [x] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [x] I tested locally to make sure this feature/fix works
- [x] I have run `yarn type-check` & `yarn build` to confirm there are not any associated errors
- [ ] This PR passes the Circle CI checks

### CODE REVIEW

File: examples/with-nextjs/public/vercel.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg width="283" height="64" viewBox="0 0 283 64" fill="none"
```
Comment (Adamj1232) [inline examples/with-nextjs/public/vercel.svg:1]: @taylorjdawson do we want to remove this?

File: examples/with-nextjs/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@web3-onboard/react": "file:.yalc/@web3-onboard/react",
```
Comment (mahmud-bn) [inline examples/with-nextjs/package.json:12]: I guess this is leftover from testing.

Comment (taylorjdawson) [reply examples/with-nextjs/package.json:12]: Yeah it is, good catch!

File: packages/react/package.json [hunk 3]
```diff
@@ -50,20 +50,23 @@
     "dist"
   ],
   "scripts": {
-    "build": "tsc",
+    "build": "yarn clean && tsc",
+    "clean": "rimraf dist",
+    "prepublishOnly": "yarn build",
```
Comment (Adamj1232) [inline packages/react/package.json:55]:
```suggestion
    "prepublish-only": "yarn build",
```
Kabob case to conform with other commands?

File: examples/with-nextjs/public/vercel.svg (hunk 1 above)
Comment (Adamj1232) [reply examples/with-nextjs/public/vercel.svg:1]: Not required but always nice to remove unnecessary files/code

File: packages/react/package.json (hunk 3 above)
Comment (taylorjdawson) [reply packages/react/package.json:55]: Oh that's the yarn lifecycle hook that gets called during yarn publish script and I think it need to be camel case?

**Pull request closed without merging**